zerver/views/registration.py

```diff
@@ -51,6 +53,12 @@
 def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -> HttpResponse:
     # If the key isn't valid, show the error message on the original URL
     confirmation = Confirmation.objects.filter(confirmation_key=confirmation_key).first()
+    # if the prereg user has status revoked the confirmation link should throw an error page
+    confirmation_status_obj = Confirmation.objects.get(confirmation_key=confirmation_key,
+                                                       type=confirmation.type)
```

Comment:
I'm confused by this query, why are we doing it? How does it relate to the query above and why does it use `type=confirmation.type` :thinking:

Reply:
hmm indeed. I don't know too. I checked on git blame if I did this because I don't remember doing it. I'm the only author on the commit in my branch, but I entered on Arunika's branch and found the original commit where she's the author. I'll try to understand why she did this.